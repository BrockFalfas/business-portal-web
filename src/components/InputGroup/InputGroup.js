import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './InputGroup.scss';

export class InputGroup extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    errors: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    label: PropTypes.string,
  };

  render() {
    const { className, children, errors, label } = this.props;

    return (
      <div className={classNames('InputGroup', className)}>
        <div className="InputGroup-label">{label}</div>
        {React.Children.map(children, child =>
          React.cloneElement(child, { className: 'InputGroup-input' })
        )}
        {errors && <div className="InputGroup-errors">{errors}</div>}
      </div>
    );
  }
}

export default InputGroup;
