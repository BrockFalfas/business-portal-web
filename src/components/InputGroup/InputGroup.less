@import "../../styles/variables.less";

.InputGroup {
  display: inline-block;
  padding: 1rem;
  vertical-align: top;

  &-label {
    text-transform: uppercase;
    font-size: 0.75rem;
    font-weight: 600;
    color: @color-grey-dark;
    letter-spacing: 0.0625rem;
  }

  &-errors {
    margin-top: 0.5rem;
    color: red;
  }

  .ant-input, .ant-select-auto-complete.ant-select .ant-input {
    color: @color-blue;
    border-color: @color-grey-medium;
    border-width: 0 0 1px 0;
    border-radius: 0;
    padding: 4px 0 0;
    height: 36px;
    background: transparent;

    &:focus, &:hover, &:active, &:focus-within, &:visited {
      outline: none !important;
      border-top-color: transparent !important;
      border-left-color: transparent !important;
      border-right-color: transparent !important;
      box-shadow: none;
    }
  }

  .ant-calendar-picker-icon {
    top: 50%;
    margin-top: 0;
    transform: translateY(-50%);
  }

  .ant-select-auto-complete.ant-select .ant-select-selection__placeholder{
    margin: -10px 0 0 0;
  }

  .ant-select-auto-complete.ant-select .ant-select-selection__rendered {
    line-height: 34px;
  }

}
