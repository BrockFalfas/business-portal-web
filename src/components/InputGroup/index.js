import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './InputGroup.css';

export class InputGroup extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    errors: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    label: PropTypes.string,
  };

  render() {
    const { className, children, errors, label } = this.props;

    return (
      <div className={classNames('InputGroup', className)}>
        <div className="InputGroup-label">{label}</div>
        {children}
        {errors && <div className="InputGroup-errors">{errors}</div>}
      </div>
    );
  }
}

export default InputGroup;
