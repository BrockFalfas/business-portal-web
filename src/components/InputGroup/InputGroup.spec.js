import React from 'react';
import { shallow } from 'enzyme';

import { InputGroup } from './InputGroup';

describe('component: <InputGroup/>', () => {
  describe('InputGroup', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<InputGroup />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render children properly', () => {
      const fakeChild = <div>fake child</div>;
      const wrapper = shallow(<InputGroup>{fakeChild}</InputGroup>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
