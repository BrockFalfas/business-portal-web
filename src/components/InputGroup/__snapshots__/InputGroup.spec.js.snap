// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <InputGroup/> InputGroup should render children properly 1`] = `
<div
  className="InputGroup"
>
  <div
    className="InputGroup-label"
  />
  <div
    className="InputGroup-input"
    key=".0"
  >
    fake child
  </div>
</div>
`;

exports[`component: <InputGroup/> InputGroup should render without crashing 1`] = `
<div
  className="InputGroup"
>
  <div
    className="InputGroup-label"
  />
</div>
`;
